automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/analysis/MethodRedactedSpec.scala

```diff
@@ -0,0 +1,168 @@
+package org.broadinstitute.dsde.firecloud.test.analysis
+
+import java.util.UUID
+
+import org.broadinstitute.dsde.firecloud.component.MessageModal
+import org.broadinstitute.dsde.firecloud.fixture.{TestData, UserFixtures}
+import org.broadinstitute.dsde.firecloud.page.workspaces.methodconfigs.{WorkspaceMethodConfigDetailsPage, WorkspaceMethodConfigListPage}
+import org.broadinstitute.dsde.workbench.auth.AuthToken
+import org.broadinstitute.dsde.workbench.config.{Config, Credentials, UserPool}
+import org.broadinstitute.dsde.workbench.fixture._
+import org.broadinstitute.dsde.workbench.service.test.WebBrowserSpec
+import org.scalatest._
+
+class MethodRedactedSpec extends FreeSpec with Matchers with WebBrowserSpec with WorkspaceFixtures with UserFixtures with MethodFixtures with BillingFixtures {
+
+  val methodConfigName: String = SimpleMethodConfig.configName + "_" + UUID.randomUUID().toString
+
+  "For a method config that references a redacted method" - {
+    "should be able to choose new method snapshot" in withWebDriver { implicit driver =>
+      val user: Credentials = UserPool.chooseProjectOwner
+      implicit val authToken: AuthToken = user.makeAuthToken()
+      withCleanBillingProject(user) { billingProject =>
+        withWorkspace(billingProject, "MethodRedactedSpec_choose_new_snapshot") { workspaceName =>
+          withMethod("MethodRedactedSpec_choose_new_snapshot", MethodData.SimpleMethod, 2) { methodName =>
+            api.methodConfigurations.createMethodConfigInWorkspace(billingProject, workspaceName, MethodData.SimpleMethod.copy(methodName = methodName),
```

Comment:
if you add this here, you won't need to repeat the `copy` later:
`val method = MethodData.SimpleMethod.copy(methodName = methodName)`
and then you also won't need to reference `SimpleMethod` again.